crates/extensions/aws_mapper_ext/src/converter.rs

```diff
@@ -24,13 +24,13 @@ pub struct AwsConverter {
 }
 
 impl AwsConverter {
-    pub fn new(add_timestamp: bool, clock: Box<dyn Clock>) -> Self {
+    pub fn new(add_timestamp: bool, clock: Box<dyn Clock>, mqtt_schema: &MqttSchema) -> Self {
```

Comment:
```suggestion
    pub fn new(add_timestamp: bool, clock: Box<dyn Clock>, mqtt_schema: MqttSchema) -> Self {
```
As we're taking a reference first and then cloning it anyway.

Reply:
fixed